runtime/lua/vim/lsp.lua

```diff
@@ -371,7 +377,7 @@ do
       local curr_lines = nvim_buf_get_lines(bufnr, 0, -1, true)
       local line_ending = buf_get_line_ending(bufnr)
       local incremental_change = sync.compute_diff(
-        cached_buffers[bufnr], curr_lines, firstline, lastline, new_lastline, client.offset_encoding or 'utf-16', line_ending)
+        cached_buffers[bufnr], curr_lines, firstline, lastline, new_lastline, client.offset_encoding, line_ending)
```

Comment:
let's keep it here and remove any fallbacks below compute_diff since compute_diff is the only thing exposed from sync.lua

Reply:
I could add it back for readability, but to be clear you want to revert all the changes made to `sync.lua` in this PR?